homeassistant/components/switch/rpi_rf.py

```diff
@@ -101,22 +105,23 @@ def is_on(self):
         """Return true if device is on."""
         return self._state
 
-    def _send_code(self, code, protocol, pulselength):
-        """Send the code with a specified pulselength."""
-        _LOGGER.info("Sending code: %s", code)
-        res = self._rfdevice.tx_code(code, protocol, pulselength)
-        if not res:
-            _LOGGER.error("Sending code %s failed", code)
-        return res
+    def _send_code(self, code_list, protocol, pulselength):
+        """Send the code(s) with a specified pulselength."""
+        _LOGGER.info("Sending code(s): %s", code_list)
+        for code in code_list:
+            self._rfdevice.tx_code(code, protocol, pulselength)
+        return True
 
     def turn_on(self):
         """Turn the switch on."""
-        if self._send_code(self._code_on, self._protocol, self._pulselength):
+        if self._send_code(self._code_on_list, self._protocol,
```

Comment:
Quite interestingly, I thought so too. It seemed that when I ran the code without the if conditional and the hardcoded return value, the switch would not turn on. I believe (but I am no expert in Python and may be wrong) if I leave out the return value entirely, then Python may execute `_send_code` in a separate thread and continue on with the rest of the lines, like setting the state etc., which means more performance will be used at the same time. I think this approach forces Python to wait for the outcome of `_send_code` before it will continue setting the state and so on, since when I changed it to be this way, the codes have been transmitted fast enough that the switch recognized them as one sequence, while the other way around it didn't.